src/azul/service/responseobjects/cart_item_manager.py

```diff
@@ -42,29 +42,36 @@ def create_cart(self, user_id, cart_name, default):
                                       key_conditions=query_dict,
                                       index_name='UserCartNameIndex') > 0:
             raise DuplicateItemError(f'Cart `{cart_name}` already exists')
-
         if default:
-            if self.dynamo_accessor.count(table_name=config.dynamo_cart_table_name,
-                                          key_conditions={'UserId': user_id},
-                                          filters={'DefaultCart': True},
-                                          index_name='UserIndex') > 0:
+            user = self.user_service.get(user_id)
+            if user['DefaultCartId'] is not None:
                 raise DuplicateItemError('Default cart already exists')
-
-        cart_id = 'default' if default else str(uuid.uuid4())
+        cart_id = str(uuid.uuid4())
         self.dynamo_accessor.insert_item(config.dynamo_cart_table_name,
-                                         item={'CartId': cart_id, 'DefaultCart': default, **query_dict})
+                                         item={'CartId': cart_id, **query_dict})
+        if default:
+            self.user_service.update(user_id, default_cart_id=cart_id)
         return cart_id
 
     def get_cart(self, user_id, cart_id):
-        return self.dynamo_accessor.get_item(config.dynamo_cart_table_name,
+        if cart_id is None:
+            user = self.user_service.get(user_id)
+            cart_id = user['DefaultCartId'] or self.create_cart(user_id, 'Default Cart', default=True)
```

Comment:
See above. 

Consider creating three methods: `get_cart` (fails if cart_id is None or cart doesn't exist) `get_default_cart` (doesn't have a `cart_id` argument and fails if default cart is absent)  and `get_or_create_default_cart` (creates the default cart if absent).

Reply:
This is now separated into three methods.